samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -54,25 +60,29 @@
 
   private final Config config;
   private final ZkBarrierForVersionUpgrade barrier;
+  private final ZkJobCoordinatorMetrics metrics;
+  private final Map<String, MetricsReporter> reporters;
 
   private StreamMetadataCache streamMetadataCache = null;
   private ScheduleAfterDebounceTime debounceTimer = null;
   private JobCoordinatorListener coordinatorListener = null;
   private JobModel newJobModel;
-
   private int debounceTimeMs;
+  long startTime;
 
   public ZkJobCoordinator(Config config) {
     this.config = config;
     ZkConfig zkConfig = new ZkConfig(config);
-    ZkKeyBuilder keyBuilder = new ZkKeyBuilder(new ApplicationConfig(config).getGlobalAppId());
+    ZkKeyBuilder keyBuilder = new ZkKeyBuilder(new ApplicationConfig(config).getAppId());
+    ReadableMetricsRegistry metricsRegistry = new MetricsRegistryMap();
```

Comment:
One suggestion (I forgot to mention before): Can you instantiate the `metricsRegistry` in the `ZkJobCoordinatorFactory` and pass it in as a constructor parameter?